vita3k/module/src/write_return_value.cpp

```diff
@@ -37,6 +37,18 @@ void write_return_value(CPUState &cpu, uint64_t ret) {
     write_reg(cpu, 1, ret >> 32);
 }
 
+void write_return_value(CPUState &cpu, std::div_t ret) {
+    write_reg(cpu, 0, ret.quot);
+    write_reg(cpu, 1, ret.rem);
+}
+
+void write_return_value(CPUState &cpu, std::lldiv_t ret) {
+    write_reg(cpu, 0, ret.quot & UINT32_MAX);
+    write_reg(cpu, 1, ret.quot >> 32);
+    write_reg(cpu, 2, ret.rem & UINT32_MAX);
+    write_reg(cpu, 3, ret.rem >> 32);
```

Comment:
The `& UINT32_MAX` is not necessary. If you decide to put it, also add it to `ret.quot >> 32` and `ret.rem >> 32` as their upper bits will be set if they are negative (they are signed integer).

Reply:
There is the same code upper for int64_t and uint64_t. So change nowhere or everywhere?

Reply:
Ah indeed, oh well I guess even though this is slightly misleading, the code is correct.